core/src/main/java/com/datastax/oss/driver/internal/core/session/DefaultSession.java

```diff
@@ -378,36 +376,27 @@ private void afterInitialNodeListRefresh(CqlIdentifier keyspace) {
                 bestVersion);
             context.getChannelFactory().setProtocolVersion(bestVersion);
 
-            // If the control connection has already initialized, force a reconnect to use the new
-            // version.
-            // (note: it might not have initialized yet if there is a custom TopologyMonitor)
-            ControlConnection controlConnection = context.getControlConnection();
-            if (controlConnection.isInit()) {
-              controlConnection.reconnectNow();
-              // Reconnection already triggers a full schema refresh
-              needSchemaRefresh = false;
-            }
+            // Note that, with the default topology monitor, the control connection is already
+            // connected with currentVersion at this point. This doesn't really matter because none
+            // of the control queries use any protocol-dependent feature.
+            // Keep going as-is, the control connection might switch to the "correct" version later
+            // if it reconnects to another node.
           }
         }
-        if (needSchemaRefresh) {
-          metadataManager
-              .refreshSchema(null, false, true)
-              .whenComplete(
-                  (metadata, error) -> {
-                    if (error != null) {
-                      Loggers.warnWithException(
-                          LOG,
-                          "[{}] Unexpected error while refreshing schema during initialization, "
-                              + "keeping previous version",
-                          logPrefix,
-                          error);
-                    }
-                  });
-        }
         metadataManager
-            .firstSchemaRefreshFuture()
-            .thenAccept(v -> afterInitialSchemaRefresh(keyspace));
-
+            .refreshSchema(null, false, true)
+            .whenComplete(
+                (metadata, error) -> {
+                  if (error != null) {
+                    Loggers.warnWithException(
+                        LOG,
+                        "[{}] Unexpected error while refreshing schema during initialization, "
+                            + "keeping previous version",
+                        logPrefix,
+                        error);
+                  }
+                  afterInitialSchemaRefresh(keyspace);
```

Comment:
Should this be invoked even in case of failure? It seems this wasn't being called in case of failure before.

Reply:
Correct, but it is, it's after the error check, not in an `else` branch.

Reply:
Correct, but it does run in all cases, it's after the error check, not in an `else` branch.